soc/xtensa/intel_adsp/common/clk.c

```diff
@@ -100,3 +100,52 @@ void adsp_clock_init(void)
 		platform_cpu_clocks[i].lowest_freq = platform_lowest_freq_idx;
 	}
 }
+
+
+#define ADSP_CLOCK_SOURCE_XTAL_OSC		0
+#define ADSP_CLOCK_SOURCE_AUDIO_CARDINAL	1
+#define ADSP_CLOCK_SOURCE_AUDIO_PLL_FIXED	2
+
+#define ADSP_CLOCK_SOURCE_MLCK_INPUT		3
+#define ADSP_CLOCK_SOURCE_WOV_RING_OSC		4
+#define ADSP_CLOCK_SOURCE_COUNT			5
+
+
+struct adsp_clock_source_desc adsp_clk_src_info[ADSP_CLOCK_SOURCE_COUNT] = {
+#ifndef CONFIG_DAI_DMIC_HW_IOCLK
+	[ADSP_CLOCK_SOURCE_XTAL_OSC] = { DT_PROP(DT_NODELABEL(sysclk), clock_frequency) },
+#else
+	/* Temporarily use the values from the configuration until set xtal frequency via ipc
+	 * support is added.
+	 */
+	[ADSP_CLOCK_SOURCE_XTAL_OSC] = { CONFIG_DAI_DMIC_HW_IOCLK },
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(audioclk), okay)
+	[ADSP_CLOCK_SOURCE_AUDIO_CARDINAL] = { DT_PROP(DT_NODELABEL(audioclk), clock_frequency) },
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(pllclk), okay)
+	[ADSP_CLOCK_SOURCE_AUDIO_PLL_FIXED] = { DT_PROP(DT_NODELABEL(pllclk), clock_frequency) },
+#endif
+	[ADSP_CLOCK_SOURCE_MLCK_INPUT] = { 0 },
+#ifdef ADSP_CLOCK_HAS_WOVCRO
+	[ADSP_CLOCK_SOURCE_WOV_RING_OSC] = { DT_PROP(DT_NODELABEL(sysclk), clock_frequency) },
```

Comment:
Humm, is this intentional that the frequency will be the same as the xtal_osc? That makes no sense to me.

Reply:
This is correct. At the moment the same clock source is used, but this may change in the future.